packages/backend/src/apps/lettersg/helpers/attachment.ts

```diff
@@ -0,0 +1,29 @@
+import { type IGlobalVariable } from '@plumber/types'
+
+import { COMMON_S3_BUCKET, putObject } from '@/helpers/s3'
+
+export async function downloadAndStoreAttachmentInS3(
+  $: IGlobalVariable,
+  publicId: string,
+): Promise<string> {
+  // Note: no try catch because no known error found yet: will still throw general error
+  // separated the two http calls to avoid confusion
+  // Letters provide a redirect link so axios will auto-download it after redirecting
+  const { data: pdfData } = await $.http.get('/v1/letters/:publicId/pdfs', {
+    responseType: 'arraybuffer',
+    urlPathParams: {
+      publicId,
+    },
+    headers: {
+      Accept: 'application/pdf',
+    },
+  })
+
+  // objectKey: `${publicId}/letter`,
+  return await putObject(COMMON_S3_BUCKET, `${publicId}/letter.pdf`, pdfData, {
```

Comment:
As discussed, if got time, allow people to rename the file.

Otherwise I can make the change.

Reply:
Also pls use `https://www.npmjs.com/package/sanitize-filename` (or similar) to sanitize the filename, since we're allowing arbitrary input.

Reply:
after discussion, I will rename this to `execution-id/app-key/publicId/letter.pdf`, allow flexibility to rename the file if users request in the future